packages/catalog-server/src/lib/npm.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @overview
+ * This module contains utilities for interacting with the npm registry and
+ * downloading custom element manifests from it.
+ */
+import fetch from 'node-fetch';
+import npmFetch from 'npm-registry-fetch';
+import {
+  PackageFiles,
+  Package,
+  Version,
+} from '@webcomponents/custom-elements-manifest-tools/lib/npm.js';
+
+export class NpmAndUnpkgFiles implements PackageFiles {
+  /**
+   * Fetch package metadata from the npm registry.
+   *
+   * See https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md
+   */
+  getPackageMetadata(packageName: string): Promise<Package> {
+    return npmFetch.json(
+      `/${packageName}`
+    ) as Promise<unknown> as Promise<Package>;
```

Comment:
What's with the official types? Oh I see, they don't even see what API endpoint you're hitting, so they have no idea what type to return

Reply:
There are some packages with types for the two API endpoints, but I haven't been able to tell which is best... it does seem like this package should be typed and assume or validate that you're hitting a complaint API endpoint though.